### PR TITLE
Add  agent configuration for experimental providers

### PR DESCRIPTION
**Pull request by chwarwick**
This accepts the extra configuration needed for agent to support experimental autocomplete providers.

Includes a change from @slimsag  to properly trigger the `onDidChangeConfigurationCallback` which was responsible for creating a new `InlineCompletionItemProvider`.  

There remained an issue with the initial `InlineCompletionItemProvider` not reflecting the config sent via Init so I manually triggered the didChange event during init so it properly reflected.

### CODE REVIEW

File: agent/src/vscode-shim.ts [hunk 1]
```diff
@@ -197,7 +219,7 @@ export const commands = _commands as typeof vscode.commands
 const _env: Partial<typeof vscode.env> = {
     uriScheme: 'file',
     appRoot: process.cwd(),
-    uiKind: UIKind.Desktop,
+    uiKind: UIKind.Web,
```
Comment (olafurpg) [inline agent/src/vscode-shim.ts:222]: Just curious, was this change needed?

File: agent/src/vscode-shim.ts [hunk 2]
```diff
@@ -206,13 +228,16 @@ export let completionProvider: Promise<InlineCompletionItemProvider> = new Promi
     resolveCompletionProvider = resolve
 })
 
+let completionItemProvider: any
```
Comment (olafurpg) [inline agent/src/vscode-shim.ts:231]: This variable doesn't seem used anywhere? 🤔 I can remove it after merging.

**Review by olafurpg — APPROVED** [review]
LGTM 👍🏻 This is great progress. Nice job figuring out how everything is wired together with almost no guidance 👏🏻  There's a lot going on in this PR and the JetBrains PR

File: agent/src/vscode-shim.ts (hunk 1 above)
Comment (chwarwick) [reply agent/src/vscode-shim.ts:222]: I don't think it is required, the original issue was `UIKind` not being shimmed (I pushed that straight to your branch) due to some new changes in the VS Code extension and then when trying to figure out why the `CompletionItemProvider` wasn't properly getting the right provider I set it and flipped it.

File: agent/src/vscode-shim.ts (hunk 2 above)
Comment (chwarwick) [reply agent/src/vscode-shim.ts:231]: It looks like it can be removed,  that was my first attempt at holding a reference to the CompletionItemProvider so that it could get updated on config change, but it looks like Stephen's change worked around that.
